robolectric/src/main/java/org/robolectric/internal/BuckManifestFactory.java

```diff
@@ -33,28 +33,21 @@ public ManifestIdentifier identify(Config config) {
 
     final List<FsFile> buckResources = getDirectoriesFromProperty(buckResDirs);
     final List<FsFile> buckAssets = getDirectoriesFromProperty(buckAssetsDirs);
-
-    if (buckResources.size() != buckAssets.size()) {
-      throw new IllegalArgumentException("number of resources and assets do not match match: "
-          + buckResources.size() + " != " + buckAssets.size());
-    }
-
-    int pathCount = buckResources.size();
-    final FsFile resDir;
-    final FsFile assetsDir;
+    final FsFile resDir = buckResources.size() == 0 ? null : buckResources.get(buckResources.size() - 1);
+    final FsFile assetsDir = buckAssets.size() == 0 ? null : buckAssets.get(buckAssets.size() - 1);
     final List<ManifestIdentifier> libraries;
-    if (pathCount == 0) {
-      resDir = null;
-      assetsDir = null;
+
+    if (resDir == null && assetsDir == null) {
       libraries = null;
     } else {
-      resDir = buckResources.get(pathCount - 1);
-      assetsDir = buckAssets.get(pathCount - 1);
       libraries = new ArrayList<>();
 
-      for (int i = 0; i < pathCount - 1; i++) {
-        libraries.add(new ManifestIdentifier(null, null,
-            buckResources.get(i), buckAssets.get(i), null));
+      for (FsFile buckResource: buckResources) {
+        libraries.add(new ManifestIdentifier(null, null, buckResource, null, null));
```

Comment:
CI is failing due to ambiguous constructor reference.  See here:
https://github.com/robolectric/robolectric/blob/master/robolectric/src/main/java/org/robolectric/internal/ManifestIdentifier.java#L17,L32